### PR TITLE
CBG-2751 each collection has its own changes ctx

### PR DESCRIPTION
**Pull request by torcolvin**
## Pre-review checklist
- [x] Removed debug logging (`fmt.Print`, `log.Print`, ...)
- [x] Logging sensitive data? Make sure it's tagged (e.g. `base.UD(docID)`, `base.MD(dbName)`)
- [x] Updated relevant information in the API specifications (such as endpoint descriptions, schemas, ...) in `docs/api`

## [Integration Tests](https://jenkins.sgwdev.com/job/SyncGateway-Integration/build?delay=0sec)
- [x] `GSI=true,xattrs=true` https://jenkins.sgwdev.com/job/SyncGateway-Integration/1572/

### CODE REVIEW

**Review by gregns1 — DISMISSED** [review]
LGTM, will allow Adam to review for anything I may have missed

File: db/blip_handler.go [hunk 1]
```diff
@@ -250,6 +247,9 @@ func (bh *blipHandler) handleSubChanges(rq *blip.Message) error {
 
 	// Ensure that only _one_ subChanges subscription can be open on this blip connection at any given time.  SG #3222.
 	collectionCtx, err := bh.collections.get(bh.collectionIdx)
+	collectionCtx.changesCtxLock.Lock()
+	defer collectionCtx.changesCtxLock.Unlock()
+
```
Comment (bbrks) [inline db/blip_handler.go]: Doing this before handling `err` seems panic-prone?

File: db/blip_handler.go [hunk 2]
```diff
@@ -380,6 +383,11 @@ func (bh *blipHandler) sendChanges(sender *blip.Sender, opts *sendChangesOptions
 	}()
 
 	base.InfofCtx(bh.loggingCtx, base.KeySync, "Sending changes since %v", opts.since)
+	collectionCtx, err := bh.collections.get(bh.collectionIdx)
+	if err != nil {
+		base.WarnfCtx(bh.loggingCtx, "[%s] Could not get collection in sendChanges: %s", err)
+		return
+	}
```
Comment (bbrks) [inline db/blip_handler.go]: We've already done this (or have the `blipSyncCollectionContext` available) in the callers of `sendChanges`, so maybe we store `collectionCtx.changesCtx` in `sendChangesOptions` given `bh.collections.get` would be doing locking/map lookup work twice in most cases for this.

**Review by bbrks — CHANGES_REQUESTED** [review]
Seems like an appropriate fix 👍 See inline for a few specific comments.

**Review by bbrks — APPROVED** [review]

**Pull request merged**